tests/AblyRestTest.php

```diff
@@ -222,7 +222,7 @@ public function testCustomHostAndFallbacks() {
             $this->fail('Expected the request to fail');
         } catch(AblyRequestException $e) {
            $this->assertEquals( $hostWithFallbacks[0], $ably->http->failedHosts[0], 'Expected to try restHost first' );
-            $this->assertNotEquals( $hostWithFallbacks, $ably->http->failedHosts, 'Expected to have fallback hosts randomized' );
+            // $this->assertNotEquals( $hostWithFallbacks, $ably->http->failedHosts, 'Expected to have fallback hosts randomized' ); // this may fail when randomized order matches the original order
```

Comment:
Yup, we have approached this problem quite differently in other langauges, often just checking that the random method is just called.